PluginsAndFeatures/azure-toolkit-for-intellij/src/com/microsoft/azure/hdinsight/spark/run/configuration/RemoteDebugRunConfiguration.java

```diff
@@ -98,16 +97,35 @@ public void checkConfiguration() throws RuntimeConfigurationException {
     public RunProfileState getState(@NotNull Executor executor, @NotNull ExecutionEnvironment executionEnvironment) throws ExecutionException {
         SparkBatchJobSubmissionState state = new SparkBatchJobSubmissionState(getProject(), jobModel);
 
-        state.createAppInsightEvent(executor, new HashMap<>());
+        state.createAppInsightEvent(executor, actionProperties.entrySet().stream().collect(Collectors.toMap(
+                (Map.Entry<Object, Object> entry) -> (String) entry.getKey(),
+                (Map.Entry<Object, Object> entry) -> (String) entry.getValue()
```

Comment:
I think Properties can be converted HashMap directly

Reply:
In fact, no. 😄